functions/pollQueue.js

```diff
@@ -0,0 +1,139 @@
+"use strict";
+
+const AWS = require("aws-sdk");
+const DBD = new AWS.DynamoDB.DocumentClient();
+const SQS = new AWS.SQS({ apiVersion: "2012-11-05" });
+const Lambda = new AWS.Lambda({ apiVersion: "2015-03-31" });
+
+const {
+  CONFIG_TABLE,
+  QUEUE_NAME,
+  CONTENT_BUCKET,
+  PROCESS_QUEUE_FUNCTION
+} = process.env;
+
+const RATE_LIMIT = 5;
+const RATE_PERIOD = 1000;
+const MAX_LONG_POLL_PERIOD = 20;
+const POLL_DELAY = 100;
+const EXECUTION_MUTEX_KEY = "pollQueueExecutionExpires";
+const EXECUTION_MUTEX_TTL = 50 * 1000;
+
+const wait = delay => new Promise(resolve => setTimeout(resolve, delay));
+
+// Running list of timestamps for hits on rate limit
+let rateHits = [];
+
+module.exports.handler = async function(event, context) {
+  const now = Date.now();
+
+  try {

```

Comment:
This is a rough attempt at keeping the poller from being run twice concurrently. *Usually* the cloudwatch alarm goes off only once per minute. Every now and then it can go off as many as 3 times.